test-apps/imodel-transformer/src/Transformer.ts

```diff
@@ -63,6 +68,27 @@ export class Transformer extends IModelTransformer {
     Logger.logInfo(progressLoggerCategory, `numSourceRelationships=${this._numSourceRelationships}`);
   }
 
+  /** Initialize IModelTransformer to exclude SubCategory Elements and geometry entries in a SubCategory from the target iModel.
+   * @param subCategoryNames Array of SubCategory names to exclude
+   * @note This sample code assumes that you want to exclude all SubCategories of a given name regardless of parent Category
+   */
+  private excludeSubCategories(subCategoryNames: string[]): void {
+    const sql = `SELECT ECInstanceId FROM ${SubCategory.classFullName} WHERE CodeValue=:subCategoryName`;
+    for (const subCategoryName of subCategoryNames) {
+      this.sourceDb.withPreparedStatement(sql, (statement: ECSqlStatement): void => {
+        statement.bindString("subCategoryName", subCategoryName);
+        while (DbResult.BE_SQLITE_ROW === statement.step()) {
+          const subCategoryId = statement.getValue(0).getId();
+          this.context.filterSubCategory(subCategoryId); // filter out geometry entries in this SubCategory from the target iModel
+          this.exporter.excludeElement(subCategoryId); // exclude the SubCategory Element itself from the target iModel

```

Comment:
Does this allow the default subcategory to be excluded?

Reply:
The native code would not have filtered out geometry entries in the default SubCategory. However, an explicit check on the TypeScript side makes the intent more clear, so I have added that.